File: app/src/main/java/com/example/test1/Retrofit/APIClient.java
package com.example.test1.Retrofit;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class APIClient {
  private static Retrofit retrofit;
  private static final String BASE_URL ="https://test-es-api.hahalolo.com/";

  public static APIService getClient() {
    if (retrofit == null) {
      retrofit = new Retrofit
          .Builder()
          .baseUrl(BASE_URL)
          .addConverterFactory(GsonConverterFactory.create())
          .build();
    }

    return retrofit.create(APIService.class);
  }
}

File: app/src/main/java/com/example/test1/Retrofit/APIClient.kt
package com.example.test1.Retrofit

import com.jakewharton.retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory


object APIClient {
    private var retrofit: Retrofit? = null
    private const val BASE_URL = "https://test-es-api.hahalolo.com/"
    val client: APIService
        get() {
            if (retrofit == null) {
                retrofit = Retrofit.Builder()
                        .baseUrl(BASE_URL)
                        .addConverterFactory(GsonConverterFactory.create())
                        .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                        .build()
            }
            return retrofit!!.create(APIService::class.java)
        }
}